src/components/KDateTimePicker/KDateTimePicker.spec.ts

```diff
@@ -0,0 +1,237 @@
+    cy.getTestId(timepickerInput).click()
+    cy.get('.vc-pane-container').should('exist')
+    cy.get('.vc-pane-container').find('.vc-time-select').should('exist')
+    cy.getTestId(timepickerInput).find('.timepicker-display').should('contain.text', todayDateTimeString)
+  })
+
+  it('renders a range date and time picker', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'dateTime',
+        modelValue: defaulTimeRange,
+        range: true,
+      },
+    })
+    cy.getTestId(timepickerInput).should('exist')
+    cy.getTestId(timepickerInput).click()
+    cy.get('.vc-pane-container').should('exist')
+    cy.get('.vc-pane-container').find('.vc-time-select').should('exist')
+  })
+
+  it('displays a valid date when "Submit" is clicked', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'date',
+        modelValue: today,
+        range: false,
+      },
+    })
+
+    // Open the date time picker, click "Submit"
+    cy.getTestId(timepickerInput).click()
+    cy.get('.k-popover-content').should('be.visible')
+    cy.getTestId(clearButton).should('exist')
+    cy.getTestId(submitButton).should('exist')
+  })
+
+  it('disables "Apply" button when "Clear" is clicked', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'date',
+        modelValue: today,
+        range: false,
+      },
+    })
+
+    // Open the date time picker, click "Clear"
+    cy.getTestId(timepickerInput).click()
+    cy.get('.k-popover-content').should('be.visible')
+    cy.getTestId(clearButton).should('exist')
+    cy.getTestId(submitButton).should('exist')
+
+    // "Apply" button is disabled after selection is cleared
+    cy.getTestId(clearButton).click({ force: true, multiple: true })
+    cy.getTestId(submitButton).invoke('attr', 'disabled').should('eq', 'disabled')
+  })
+
+  it('displays todays date after clicking "Apply', () => {
+    mount(KDateTimePicker, {
+      props: {
+        // testMode: true,
+        mode: 'date',
+        modelValue: today,
+        range: false,
+      },
+    })
+
+    // Open the date time picker, click "Submit"
+    cy.getTestId(timepickerInput).click()
+    cy.getTestId(submitButton).click({ force: true, multiple: true })
+
+    // Check emitted raw date value, and the displayed value - should be the same, if rounded
+    cy.getTestId(timepickerInput).find('.timepicker-display').should('contain.text', todayDateString)
+  })
+
+  it('renders relative time frames, and makes a selection', () => {
+    mount(KDateTimePicker, {
+      props: {
+        mode: 'relative',
+        modelValue: today,
+        range: true,
+        timePeriods: exampleTimeFrames,
+      },
+    })
+
+    cy.getTestId(timepickerInput).click()
+    cy.get('.timeframe-section').should('exist')
+    cy.get('.timeframe-buttons').should('exist')
+
+    // Click on "3 hours", check whether selected class is applied
+    cy.getTestId('select-timeframe-10800000').click({ force: true, multiple: true })
```

Comment:
I'm not sure you should use force as it will click even if disabled

Reply:
All of the time frame buttons are enabled (and clickable) by default. This _would_ be a dangerous thing to do if the time picker was running in dual mode (eg: both Custom + Relative tabs), but that's not the case here.